objects/src/accounts/storage/header.rs

```diff
@@ -0,0 +1,132 @@
+use alloc::vec::Vec;
+
+use vm_core::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use vm_processor::DeserializationError;
+
+use super::{AccountStorage, StorageSlotType, Word};
+use crate::AccountError;
+
+// ACCOUNT STORAGE HEADER
+// ================================================================================================
+
+/// Account storage header is a lighter version of the [AccountStorage] storing
+/// only the [StorageSlotType] and associated values of the [super::StorageSlot]s
+/// contained in the storage.
+///
+/// The use of a header is useful in the situation where the storage is heavy (i.g. multiple Mb's),
+/// and should be enough to execute most transactions.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AccountStorageHeader {
+    slots: Vec<(StorageSlotType, Word)>,
+}
+
+impl AccountStorageHeader {
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+
+    // Returns a new instance of account storage header initialized with the provided slots.
+    pub fn new(slots: Vec<(StorageSlotType, Word)>) -> Self {
+        Self { slots }
+    }
```

Comment:
Let's add an `assert!` here to make sure we don't exceed the number of allowed storage slots (and document this in the comment in the `# Panics` section).